bc_obps/service/report_additional_data.py

```diff
@@ -1,24 +1,73 @@
-from registration.models import Operation
-from reporting.models import ReportVersion, Report, ReportAdditionalData
-from reporting.schema.report_additional_data import ReportAdditionalDataIn
+from typing import Optional
+
+from reporting.models import ReportVersion, ReportAdditionalData
+from reporting.schema.report_additional_data import (
+    ReportAdditionalDataOut,
+    CaptureEmissionSchema,
+    AdditionalDataSectionSchema,
+    ReportAdditionalDataIn,
+)
 
 
 class ReportAdditionalDataService:
     @staticmethod
-    def get_report_person_responsible_by_version_id(
+    def get_report_report_additional_data(
         report_version_id: int,
-    ) -> ReportAdditionalData:
-        return ReportAdditionalData.objects.get(report_version__id=report_version_id)
+    ) -> Optional[ReportAdditionalDataOut]:
+        # Fetch the report additional data for the given report version ID
+        report_additional_data = ReportAdditionalData.objects.filter(report_version__id=report_version_id).first()
+
+        if report_additional_data is None:
+            return None
+
+        # Determine the capture type based on the attributes of report additional data
+        capture_type = []
+        if report_additional_data.emissions_on_site_use is not None:
+            capture_type.append("On-site use")
+        if report_additional_data.emissions_on_site_sequestration is not None:
+            capture_type.append("On-site sequestration")
+        if report_additional_data.emissions_off_site_transfer is not None:
+            capture_type.append("Off-site transfer")
+
+        # Fetch the registration purpose of the associated report
+        registration_purpose = (
+            ReportVersion.objects.select_related("report", "report__operation")
+            .get(id=report_version_id)
+            .report.operation.registration_purpose
+        )
+
+        # Prepare the additional data section if the conditions are met
+        additional_data_section = (
+            {"electricity_generated": report_additional_data.electricity_generated}
+            if registration_purpose == "OBPS Regulated Operation" and report_additional_data.electricity_generated
+            else None
+        )
+
+        # Construct and return a ReportAdditionalDataOut instance
+        return ReportAdditionalDataOut(
```

Comment:
All this needs to be done in a django ninja schema

Your schema will also be a lot easier to test than this (I don't think there are tests for this service) since we won't have to test the field transformations, just the fact that it's setup properly